src/XrdHttpTpc/XrdHttpTpcTPC.cc

```diff
@@ -407,11 +407,30 @@ int TPCHandler::GetContentLengthTPCPull(CURL *curl, XrdHttpExtReq &req, uint64_t
     contentLength = state.GetContentLength();
     return result;
 }
-  
+
 /******************************************************************************/
 /*            T P C H a n d l e r : : S e n d P e r f M a r k e r             */
 /******************************************************************************/
-  
+
+int TPCHandler::SendPerfMarker(XrdHttpExtReq &req, TPCLogRecord &rec, TPC::State &state, std::string desc) {
+    std::stringstream ss;
+    const std::string crlf = "\n";
+    ss << "Perf Marker" << crlf;
+    ss << "Timestamp: " << time(NULL) << crlf;
+    ss << "Stripe Index: 0" << crlf;
+    ss << "Stripe Bytes Transferred: " << state.BytesTransferred() << crlf;
+    ss << "Total Stripe Count: 1" << crlf;
+    if (!desc.empty()) ss << "RemoteConnections: " << desc << crlf;
+    ss << "End" << crlf;
+    rec.bytes_transferred = state.BytesTransferred();
+    logTransferEvent(LogMask::Debug, rec, "PERF_MARKER");
+    return req.ChunkResp(ss.str().c_str(), 0);
+}
```

Comment:
This is adding code duplication, we already have two `TPCHandler::SendPerfMarker` methods. We should not add a third. I think it's better to think how to consolidate them into a single one instead.

Also, this commit is mixing up a lot more changes than what it says it's doing (`Update remote connection list from curl`). Please try to structure changes as one logical change per commit.